lib/pages/settings.dart

```diff
@@ -126,6 +153,137 @@ class _AccountsConfig extends StatelessWidget {
   }
 }
 
+class _AccountsConfigAddInstanceDialog extends HookWidget {
+  final GlobalKey<ScaffoldState> scaffoldKey;
+
+  const _AccountsConfigAddInstanceDialog({@required this.scaffoldKey})
+      : assert(scaffoldKey != null);
+
+  @override
+  Widget build(BuildContext context) {
+    var instanceController = useTextEditingController();
+    useValueListenable(instanceController);
+
+    var loading = useState(false);
+
+    return AlertDialog(
+      title: Text('Add instance'),
+      content: TextField(
+        autofocus: true,
+        controller: instanceController,
+        decoration: InputDecoration(
+          border: OutlineInputBorder(),
+          labelText: 'Instance url',
+        ),
+      ),
+      actions: <Widget>[
+        FlatButton(
+          child: !loading.value ? Text('Add') : CircularProgressIndicator(),
+          onPressed: instanceController.text.isEmpty
+              ? null
+              : () async {
+                  try {
+                    loading.value = true;
+                    await context
+                        .read<AccountsStore>()
+                        .addInstance(instanceController.text);
+                    scaffoldKey.currentState.hideCurrentSnackBar();
+                  } on Exception catch (err) {
+                    scaffoldKey.currentState.showSnackBar(SnackBar(
+                      content: Text(err.toString()),
+                    ));
+                  }
+                  loading.value = false;
+                  Navigator.of(context).pop();
+                },
```

Comment:
should be separated to a function. not be directly in UI. can be in the build method, just not in the UI